cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -122,6 +130,41 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     String subnet = conf.getSubnet();
     Network networkInfo = getNetworkInfo(projectId, network, compute);
 
+    PeeringState state = getPeeringState(systemNetwork, systemProjectId, networkInfo, compute);
+
+    if (privateInstance) {
+      // private instance must have a peering established
+      verifyPeering(conf, network, systemNetwork, projectId, systemProjectId, state);
+      if (conf.isPreferExternalIP()) {
+        // When prefer external IP is set to true it means only Dataproc external ip can be used to for communication
+        // the instance being private instance is incapable of using external ip for communication
+        throw new IllegalArgumentException("The instance is incapable of using external ip for communication " +
+                                             "with Dataproc cluster. Please correct profile configuration by " +
+                                             "deselecting preferExternalIP.");
+      }
+    } else if (conf.isPreferExternalIP() && state == PeeringState.ACTIVE) {
+      // Peering is setup between the system network and customer network and is in ACTIVE state.
+      // However user has selected to preferred external IP the instance. This is not a private instance and is
+      // capable of communicating with Dataproc cluster with external ip so just add warning message indicating that
+      // internal IP can be used.
+      LOG.info(String.format("VPC Peering from network '%s' in project '%s' to network '%s' " +
+                               "in project '%s' is in the ACTIVE state. Prefer External IP can be set to false " +
+                               "to launch Dataproc clusters with internal IP only.", systemNetwork,
+                             systemProjectId, network, projectId));
+    }
+
+    boolean useInternalIP;
+    if (privateInstance) {
```

Comment:
I don't think this if-else isn't needed, it's logically equivalent to just having the else clause.